rust/chains/hyperlane-ethereum/src/mailbox.rs

```diff
@@ -178,13 +185,28 @@ where
     /// Create a reference to a mailbox at a specific Ethereum address on some
     /// chain
     pub fn new(provider: Arc<M>, locator: &ContractLocator) -> Self {
+        // Arbitrum Nitro based chains are a special case for transaction cost estimation.
+        // The gas amount that eth_estimateGas returns considers both L1 and L2 gas costs.
+        // We use the NodeInterface to isolate the L2 gas costs.
+        // See https://developer.arbitrum.io/arbos/gas#nodeinterfacesol or https://github.com/OffchainLabs/nitro/blob/master/contracts/src/node-interface/NodeInterface.sol#L110
+        let arbitrum_node_interface = if locator.domain.is_arbitrum_nitro() {
+            // Arbitrum's NodeInterface is accessible at address(0xC8)
+            Some(Arc::new(ArbitrumNodeInterface::new(
+                H160::from_str("0x00000000000000000000000000000000000000C8").unwrap(),
```

Comment:
feels a little janky to be using from_str, wonder if you feel there's a more elegant way @mattiecnvr

Reply:
I probably would have written something similar but this is a way you _could_ write it. (Just let cargo fmt figure out how to make it pretty)

```rs
let arbitrum_node_interface = locator.domain.is_arbitrum_nitro().then(|| Arc::new(ArbitrumNodeInterface::from_low_u64_ne(0xC8), provider.clone())))
```

Reply:
much nicer, ty